tests/test_engine/test_update/test_update.py

```diff
@@ -320,6 +320,62 @@ def test_update_positional(monty_update, mongo_update):
     assert next(monty_c) == {"a": [{"b": 3, "c": 1}, {"b": 5, "c": 0}]}
 
 
+def test_update_positional_2(monty_update, mongo_update):
+    docs = [
+        {"users": [{"id": 1, "name": None}, {"id": 2, "name": None}]}
+    ]
+    find = {"users.id": 2}
+    spec = {"$set": {"users.$.name": "david"}}
+
+    mongo_c = mongo_update(docs, spec, find)
+    monty_c = monty_update(docs, spec, find)
+
+    assert next(mongo_c) == next(monty_c)
+    monty_c.rewind()
+    assert next(monty_c) == {
+        "users": [{"id": 1, "name": None}, {"id": 2, "name": "david"}],
+    }
```

Comment:
This one test actually passed, because the positional update operation is working under same field as query `users`.